src/main/kotlin/com/beust/kobalt/app/BuildFileCompiler.kt

```diff
@@ -95,6 +95,7 @@ public class BuildFileCompiler @Inject constructor(@Assisted("buildFiles") val b
             val taskResult = maybeCompileBuildFile(context, BuildFile(Paths.get(modifiedBuildFile.path),
                     "Modified ${Constants.BUILD_FILE_NAME}", buildFile.realPath),
                     buildScriptJarFile, pluginUrls)
+            modifiedBuildFile.delete()
```

Comment:
well, this was my first solution too, but what about when kobalt is running in server mode?